gobblin-metastore/src/main/java/gobblin/metastore/DatabaseJobHistoryStore.java

```diff
@@ -318,296 +271,440 @@ public void close()
     // Nothing to do
   }
 
+  private static ImmutablePair<MigrationVersion, MigrationVersion> getDatabaseVersions(DataSource dataSource)
```

Comment:
can you break this down into two methods `getLatestDatabaseVersion` and `getCurrentDatabaseVersion` I know it may require a redundant call to `flyway.info()` but it makes the code more readable.